### PR TITLE
fix: update telemetry dep to fix mapbox rn crash

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **fix**

This PR resolves [CX-2153]
This PR resolves [CX-2154]

### Description

A crash was occurring on pages showing maps on Android.

The issue was using an old telemetry dependency that reads phone state without permission: 
https://github.com/react-native-mapbox-gl/maps/issues/1286

Fix is to update the telemetry dependency in our own build.gradle

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Fix a crash on pages showing maps - Brian

#### Dev changes

-

</details>

[CX-2153]: https://artsyproduct.atlassian.net/browse/CX-2153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-2154]: https://artsyproduct.atlassian.net/browse/CX-2154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Android user-facing changes (Fix a crash on pages showing maps - Brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d6a01fdf314b137e8f7766ccddb49f24a2482fda
</p>

**Review by araujobarret — APPROVED** [review]

File: android/app/build.gradle [hunk 1]
```diff
@@ -223,6 +223,9 @@ dependencies {
     implementation "com.google.firebase:firebase-analytics:19.0.0" // so we can see push notification reports on firebase
     implementation 'com.google.firebase:firebase-messaging:21.1.0'
 
+    // workaround for crash here: https://github.com/react-native-mapbox-gl/maps/pull/1376
+    implementation 'com.mapbox.mapboxsdk:mapbox-android-telemetry:6.1.0'
```
Comment (araujobarret) [inline android/app/build.gradle:227]: Maybe adding the reference to `HACKS.md`?

**Review by lordkiz — APPROVED** [review]
This can serve as a workaround for now.
However, the latest `react-native-mapbox/maps` already has [included this as a dependency](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/build.gradle#L58). So bumping the package to v8.4.0 should be enough.
Also, we don't want to do the bumping now because we might introduce unknown bugs into iOS. So I will create a ticket for that.
Otherwise, this looks good to me.

**Comment by lordkiz** [comment]
Follow up work: https://artsyproduct.atlassian.net/browse/CX-2159

File: android/app/build.gradle (hunk 1 above)
Comment (lordkiz) [reply android/app/build.gradle:227]: @araujobarret we'll do the cleanup here: https://artsyproduct.atlassian.net/browse/CX-2159

**Pull request merged**